vispy/visuals/volume.py

```diff
@@ -495,6 +548,7 @@ def set_data(self, vol, clim=None, copy=True):
             Colormap limits to use. None will use the min and max values.
         copy : bool | True
             Whether to copy the input volume prior to applying clim normalization.
+            Has no effect if visual was created with 'texture_format' not equal to None.
```

Comment:
This bit is surprising. How do we update a visual if we want to change the data? In napari, for example, when we move the slider, e.g. to get a different timepoint of a 3D volume, we call `.set_data()`. We would still want to do this even if we are using this new zero-CPU-copy method.

... And anyway from my reading of the code below, this method *does* update the texture value even in that case?

Reply:
@jni, this isn't so much about whether the visual gets update by `.set_data()` (it always does).  It's whether the data array needs to be copied before scaling it prior to GPU upload (so as not to mutate the input data).  You may recall in https://github.com/napari/napari/issues/575#issuecomment-544753893, we learned that the `VolumeVisual` was mutating data.  So this copy step was added in https://github.com/vispy/vispy/commit/876a0d3 ... but if the raw data is uploaded to the GPU texture... it's no longer needed! 🎉

@djhoese  ... this also reminds me: technically, we no longer need to retain the `self._last_data` pointer if `self._scale_texture_gpu is True`, since it will never be used in that case.

Reply:
@jni to clarify, this "Has no effect" statement refers only to the `copy` keyword argument, not `set_data` as a whole. I'll reword some of it.

Reply:
Updated.

Reply:
Right, sorry for the noise, the newline messed with my brain's numpydoc parser. 😂 🤦 😂